cmake/traccc-compiler-options-hip.cmake

```diff
@@ -0,0 +1,12 @@
+# TRACCC library, part of the ACTS project (R&D line)
+#
+# (c) 2021-2024 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+# Include the helper function(s).
+include( traccc-functions )
+
+#HIP requires position-independent executables for linking
+traccc_add_flag( CMAKE_HIP_FLAGS "-fPIE" )
```

Comment:
I'm on board with adding a placeholder file like this, but `-fPIE` should not be added like this. Please use the following formalism instead:

https://github.com/acts-project/vecmem/blob/main/tests/hip/CMakeLists.txt#L30-L34

That is just more platform independent.

Reply:
No. Rather, you should add

```cmake
set_target_properties( traccc_test_hip PROPERTIES
   POSITION_INDEPENDENT_CODE TRUE )
```

in the newly introduced HIP test `CMakeLists.txt`.

You see, shared libraries are **always** built with "position independent code". (Otherwise they would fundamentally not work.) It's static libraries and executables that usually are built with position **dependent** code. So we just need to tell a couple of them to "be independent". We shouldn't do this on a global scale at all.

Reply:
Ah, I misunderstood. Added this in the two executables that need it.